src/gui.cc

```diff
@@ -541,17 +541,20 @@ void GUI::DisplayTimeline(
             bool item_present = false;
             TogglTimelineEventView *event_app = first_event;
             while (event_app) {
-                if (compare_string(event_app->Filename, copy_string(event.Filename().c_str())) == 0) {
+                char_t *filename = copy_string(event.Filename().c_str());
```

Comment:
While this fixes the memory leak, it adds an unnecessary `strdup`/`free` pair of calls, increasing system load. I outlined how to fix this properly in the comments at the previous commit.

Reply:
I understand, but as I mention that I would choose the safe approach this PR.

I suggest that we should move the actually fix to different ticket? since the fix is on `copy_string()` and it impacts the majority of the codebase as well as it's out of scope of this PR too.

If it's reasonable, we could merge then I will open the ticket 👍

Reply:
There's nothing wrong with `copy_string` itself. It's just unnecessary to use it here as it only increases the performance impact of this method.

I suggested the usage of `to_char_t`, did you try it?

Reply:
@MartinBriza As you mentioned previously, `to_char_t` is written in a dangerous way, and I can confirm this as I was writing it only for tests and, as I noticed later, missed the memory leak. So if we want to use it in production we should fix it first.